vm-monitor/src/main/java/com/epam/pipeline/vmmonitor/service/impl/GCPMonitorService.java

```diff
@@ -0,0 +1,147 @@
+        verifyProjectInfo(region);
+        final Compute compute = getGCPCompute(region);
+        final InstanceList instances = getGcpVmInstances(region, compute);
+        return createListOfVMsFromGCPInstances(instances);
+    }
+
+    @Override
+    public CloudProvider provider() {
+        return CloudProvider.GCP;
+    }
+
+
+    private void verifyProjectInfo(final GCPRegion region) {
+        Assert.notNull(region.getProject(), "Project ID is not specified for GCP region.");
+        Assert.notNull(region.getRegionCode(), "Zone ID is not specified for GCP region.");
+    }
+
+    Compute getGCPCompute(final GCPRegion region) {
+        try {
+            final GoogleCredential credential = getCredentials(region.getAuthFile());
+            return new Compute.Builder(httpTransport, jsonFactory, credential).build();
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+            throw new GCPMonitorException(e.getMessage(), e);
+        }
+    }
+
+    private GoogleCredential getCredentials(final String authFile) throws IOException {
+        if (StringUtils.isBlank(authFile)) {
+            return GoogleCredential.getApplicationDefault();
+        }
+        try (InputStream stream = new FileInputStream(authFile)) {
+            return GoogleCredential.fromStream(stream)
+                                   .createScoped(Collections.singletonList(ComputeScopes.COMPUTE_READONLY));
+        }
+    }
+
+    InstanceList getGcpVmInstances(final GCPRegion region, final Compute compute) {
+        final String filter = String.format(FILTER_PATTERN, instanceTag.getKey(), instanceTag.getValue());
+        try {
+            return Optional.ofNullable(compute.instances()
+                                              .list(region.getProject(), region.getRegionCode())
+                                              .setFilter(filter)
+                                              .execute()).orElse(new InstanceList());
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+            throw new GCPMonitorException(e.getMessage(), e);
+        }
+    }
+
+    private List<VirtualMachine> createListOfVMsFromGCPInstances(final InstanceList instances) {
+        return ListUtils.emptyIfNull(instances.getItems())
+                        .stream()
+                        .map(this::toVM)
+                        .collect(Collectors.toList());
+    }
+
+    private VirtualMachine toVM(final Instance instance) {
+        return VirtualMachine.builder()
+                             .instanceName(instance.getName())
+                             .cloudProvider(provider())
+                             .instanceId(instance.getId().toString())
+                             .privateIp(getInternalIP(instance))
+                             .tags(instance.getLabels())
+                             .build();
+    }
+
+    private String getInternalIP(Instance instance) {
```

Comment:
Missing `final` for `instance` argument

Reply:
Done [4e8957b](https://github.com/epam/cloud-pipeline/pull/403/commits/4e8957b9096e5d0eed8d990a9ecc620c039a86b7)